### PR TITLE
Docs: Limit Filters to Eligible Species

### PR DESCRIPTION
**Pull request by ax3l**
Document how to limit generic "free" filters to eligible particle species as asked in #2573.

Add to `particleFilters.param` files:

* docstring
* includes commonly needed for eligible traits
* section with prepared namespace for traits

### CODE REVIEW

File: docs/source/usage/workflows/particleFilters.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+Besides the list of pre-defined filters with parametrization users can also define generic, "free" implementations as shown above.
+All filters are added to ``AllParticleFilters`` and then *combined with all available species* from ``VectorAllSpecies`` (see :ref:`speciesDefinition.param <usage-params-core>`.
+
+In the case of user-defined free filters we can not check if each species in ``VectorAllSpecies`` fulfills the requirements of the filter.
+That means: if one accesses specific *attributes* or *flags* of a species in a filter, they must exist or will lead to a compile error.
+
+As an example, :ref:`probe particles <usage-workflows-probeParticles>` usually do not need even have a ``momentum`` attribute which would be used for an energy filter.
+So they should be ignored from compilation when combining filters with particle species.
+
+In order to exclude all species that have no ``momentum`` attribute from the ``ParticlesForwardPinhole`` filter, define the C++ trait ``SpeciesEligibleForSolver``.
```
Comment (n01r) [inline docs/source/usage/workflows/particleFilters.rst]: >define the C++ trait

is rather *specialize the C++ trait*, right?

File: docs/source/usage/workflows/particleFilters.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+Particle Filters
+----------------
+
+.. sectionauthor:: Axel Huebl
+
+A common task in both modeling, initializing and in situ processing (output) is the selection of particles of a particle species by attributes.
+PIConGPU implements such selections as *particle filters*.
+
+Particle filters are simple mappings assigning each particle of a species either ``true`` or ``false`` (ignore / filter out).
+These filters can be defined in :ref:`particleFilters.param <usage-params-core>`.
```
Comment (PrometheusPi) [inline docs/source/usage/workflows/particleFilters.rst]: QUESTION:
Is `<usage-params-core>` correct here?

File: docs/source/usage/workflows/particleFilters.rst [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+   >;
+
+   } // namespace filter
+   } // namespace particles
+   } // namespace picongpu
+
+Limiting Filters to Eligible Species
+""""""""""""""""""""""""""""""""""""
+
+Besides :ref:`the list of pre-defined filters <usage-particles-filters>` with parametrization users can also define generic, "free" implementations as shown above.
```
Comment (PrometheusPi) [inline docs/source/usage/workflows/particleFilters.rst]: GRAMMAR:
comma missing: 
`parametrization, users`

File: docs/source/usage/workflows/particleFilters.rst [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+All filters are added to ``AllParticleFilters`` and then *combined with all available species* from ``VectorAllSpecies`` (see :ref:`speciesDefinition.param <usage-params-core>`).
+
+In the case of user-defined free filters we can now check if each species in ``VectorAllSpecies`` fulfills the requirements of the filter.
+That means: if one accesses specific *attributes* or *flags* of a species in a filter, they must exist or will lead to a compile error.
+
+As an example, :ref:`probe particles <usage-workflows-probeParticles>` usually do not need even have a ``momentum`` attribute which would be used for an energy filter.
```
Comment (PrometheusPi) [inline docs/source/usage/workflows/particleFilters.rst]: TYPO:
`even` seems wrong here, did you want to write `to`?

Comment (n01r) [reply docs/source/usage/workflows/particleFilters.rst]: Nope - it explains that probe particles are a reduced particle species which is missing a lot of attributes, *even* basic ones like `momentum`.

Comment (PrometheusPi) [reply docs/source/usage/workflows/particleFilters.rst]: agree with 
do not need to have

File: docs/source/usage/workflows/particleFilters.rst (hunk 2 above)
Comment (ax3l) [reply docs/source/usage/workflows/particleFilters.rst]: Yes.

File: docs/source/usage/workflows/particleFilters.rst (hunk 4 above)
Comment (ax3l) [reply docs/source/usage/workflows/particleFilters.rst]: Yes, typo.
Add `to` or remove `have`, etc

**Comment by ax3l** [comment]
@PrometheusPi note that writing ALL CAPS is still considered shouting in "the internet" ;-)

**Comment by PrometheusPi** [comment]
@ax3l MERGED 😉

**Pull request merged**